### ISSUE TITLE
Can't set background color

### ISSUE THREAD

**Issue by Chucky109** [niqdev/ipcam-view#44]
I am not able to change the background color of the MjpegView to remove the black borders.

```
<com.github.niqdev.mjpeg.MjpegSurfaceView
                    android:id="@+id/streamView"
                    android:layout_width="match_parent"
                    android:layout_height="match_parent"
                    android:background="#FFFFFF"
                    stream:type="stream_default" />
```
Is there any magic trick or something like that?

Edit: Okay I was able to do this programmatically after casting the MjpegView into a normal view but then I have the problem that I can use every color instead of `Color.transparent`. When I try to 
`((View) streamView).setBackgroundColor(Color.TRANSPARENT);` it remains black

**Comment by niqdev** [comment]
Hi @Chucky109 , unfortunately I never tried to change the color of the background.
If you find a solution for the transparent please share it!
I would be glad if you add in the readme any info that could help also other people and if you are interested, add your project to the [list](https://github.com/niqdev/ipcam-view#apps-that-use-this-library)
Thanks

**Comment by Chucky109** [comment]
Hey @niqdev 
Unfortunately it is no public project, only for my studies. But of course I will share my solution if I find one. Will have a look on it later in the evening.
Thanks for your answer!

**Comment by Chucky109** [comment]
Okay I found something, but I'm not 100% sure if it is the key to the solution. The class `MjpegViewDefault.java` explicitly says that surface color should be black. (https://github.com/niqdev/ipcam-view/blob/dbe3d9b218554159ef71892fe4cc38ccd185d430/mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegViewDefault.java#L141)
I guess if there would be a way to configure this (setter), this could help. But I will give it a try later

**Comment by niqdev** [comment]
Yep, you could try to clone this repo and try to change that property and see if it works in the demo project.
If that work I think you could try abstract that property in `MjpegView` in such a way that you can override it from outside and leave black as default.

**Comment by niqdev** [comment]
@Chucky109 about `Okay I actually never edited a lib or something like that` you are more than welcome to make a pull request: fork the repo, clone it, create a feature branch, make your changes, make sure that you don;t introduce regression, push it and then create a PR

**Comment by Chucky109** [comment]
Okay it actually worked pretty well with the demo app - thanks for the tip 👍 I just created a PR with my changes @niqdev

**Comment by niqdev** [comment]
Well done! just merged #45 , I will close the issue when I will publish a new version of the library.
Thanks for your contribution

**Comment by Chucky109** [comment]
I'm glad that it worked out that well. First time I contribute in a library development :) Can you tell me how long it will take until the new version will be published?

**Comment by niqdev** [comment]
I've just released version `1.3.0`, enjoy!

**Comment by Chucky109** [comment]
Good morning! I made a mistake yesterday I think. I only tested it on emulator where it worked out very well. But in my application it does not work. Background stays black. I will investigate on that today and let you know when I found the solution

**Comment by niqdev** [comment]
Great, please try to verify also on different android versions if it doesn't break anything else. Thanks

**Comment by Chucky109** [comment]
Okay after several hours of looking for solutions I finally did it - I guess :D Nah, just joking. I'm pretty sure it will work this time.
The problem was the following: You can't set transparent background color to a SurfaceView. There are several workarounds. The demo app decided to show transparent color as white (maybe it's based on theme, I'm not sure). Because white and the default Activity background color aren't that much different, I didn't saw the bug. This morning, after you published 1.3.0, I updated my app and instead of showing white background it took black one. That was the moment I realised something went wrong.

I hope I could explain my solution as detailed as possible in the new readme. You now can choose whether you wan't transparent background (like me) or colored background. The first one will always overwrite the second one - but I explained that in the readme also.

I would be very thankful if you could publish it as 1.3.1 (or whatever version) so I can make sure it will also work in my app - after you reviewed of course :)

Have a nice sunday!

**Comment by niqdev** [comment]
Great, I'll review and merge your PR later!

**Comment by niqdev** [comment]
Just released version `1.3.1`. If you are happy close the issue please. I'll leave it open just in case you want to add something else.
Thanks

**Comment by Chucky109** [comment]
Very good idea :D I opened the next PR^^ In my app I show and hide the stream. This might happen multiple times. But because of the workaround to set transparency I wasn't able to hide the view. So I implemented a method to reset transparent background which basically removes the zorder property of the surface view and sets the pixelformat back to default opaque when called.

And in addition I saw that the canvas keeps the last image when I stopped and hide the stream. Therefore I implemented a possibility to clear this image.

I definately hope that this will be the final change :D By the way the transparency worked perfectly, it's just this one little detail.

**Comment by niqdev** [comment]
No worries, make as many changes as you need, as I said, for me is just important that you don't break the current functionality.
Please, can you just confirm that you test the demo app on the supported emulators and also on the real device?

**Comment by Chucky109** [comment]
Okay so I just tested it successfully. The setup was the following:
I added a button to the IpCamTwoActivity which basically executes
```java
mjpegView2.stopPlayback();
mjpegView2.resetTransparentBackground();
mjpegView2.clearStream();
```

Then I tried to hit him as often as possible, change the orientation frequently, hit the button again randomly and also turned the device off and on to ensure onPause and onResume work correctly. I also tried it without the stopPlayback (which might make no sense but it behaves as expected - clears the stream and then redraws it because it is still running).

Tested devices were:

API 16 (E), API 18 (E), API 22 (E, Real device), API 24 (E, Real device), API 26 (E)

Unfortunately I couldn't get on lower API levels because I was not able to find the emulator images therefore.
I understand your worries about this line but it is basically the same way the camera stream is drawn onto SurfaceView. I just thought about what would happen if you are able to lock the canvas while you change orientation so that he might not unlock it successful. But I think this wouldn't be a problem because the SurfaceView is recreated and therefore the init() method will run again and all that stuff. So I think there wouldn't be any problem. (I don't even know how to achieve this scenario but someone might be able to do unwanted^^)

**Comment by niqdev** [comment]
Well done! Can I ask just a favour? Would you mind to revert the changes in `IpCamTwoActivity` and move it in a new activity, named like `appearance` or `customization` and add the button for refresh directly in the navigation bar?
Maybe also look for a new public ip cam link just for your example.
In this way I will release also a new version of the demo app. Do you agree?
Thanks

**Comment by Chucky109** [comment]
Of course, no problem. One more question: What do you mean with "button for refresh"?

**Comment by Chucky109** [comment]
Oh I forgot that the PR is still open, so no need to create a new one. I pushed everything with a little additional change. The Activity in the demo app has now nearly the same functionality as in my own app with hiding and showing. And as it works in the demo app finally, I am really sure it will do the same in my app :)

**Comment by niqdev** [comment]
Great job! I will hopefully review/merge/release the PR this evening or soon anyway. Thanks

**Comment by niqdev** [comment]
@Chucky109 I released version `1.3.2` and tried to synthesize your notes.
I think the readme should be simple and coincise and it would be more helpful a detailed/verbose comment in the sample activity.
Could you please just review the readme/comments?
Thanks a lot

**Comment by Chucky109** [comment]
Okay, I will do it tomorrow 👍

**Comment by niqdev** [comment]
Sorry, my bad, just released it. I've also released version 1.6.0 of the app in the store

**Comment by Chucky109** [comment]
So I reviewed your README changes and added some more + opened the last PR. It finally works as I need it in my app. Thank you very much for your patience :D 👍

**Comment by niqdev** [comment]
merged #48 
Thanks for your contributions!
